lib/lightning/attempts/handlers.ex

```diff
@@ -160,19 +160,25 @@ defmodule Lightning.Attempts.Handlers do
     end
 
     def call(params) do
-      with {:ok, complete_run} <-
-             new(params)
-             |> apply_action(:validate) do
+      with {:ok, complete_run} <- new(params) |> apply_action(:validate) do
         update(complete_run)
       end
     end
 
     defp update(complete_run) do
       Repo.transact(fn ->
-        with run = %Run{} <-
-               get_run(complete_run.run_id) || {:error, :not_found},
+        with %Run{} = run <- get_run(complete_run.run_id),
              {:ok, _} <- to_dataclip(complete_run) |> Repo.insert() do
           update_run(run, complete_run)
+        else
+          nil ->
+            {:error,
+             %Run{}
+             |> change()
+             |> add_error(:run_id, "not found")}
```

Comment:
Please don't return a Run changeset. As mentioned in my first review:

> In the case of an error we don't return a Run changeset, we return a CompleteRun changeset

The Run model does not have a `:run_id` key. You have access to the original `CompleteRun` changeset. Please change the `nil` case to:

```elixir
{:error, complete_run |> add_error(:run_id, "not found")}
```